### ISSUE TITLE
doxygenfile directive causes duplicate IDs for sections that were not specified.

### ISSUE THREAD

**Issue by inconstxpr** [michaeljones/breathe#614]
# Behavior
I'm using the `doxygenfile` directive to display only the brief and detailed file description, like this:
```
.. doxygenfile:: DynamicArray.h
        :sections: briefdescription detaileddescription
```

I don't want the other sections, so after that, I go on to grouping functions under subheaders:
```
Lifecycle
---------
.. doxygenfunction:: DynamicArray_Create
.. doxygenfunction:: DynamicArray_CreateFromData
.. doxygenfunction:: DynamicArray_Destroy
```

However, every use of `doxygenfunction` causes a "Duplicate ID", "Duplicate explicit target name" and "Duplicate C declaration", pointing to the `doxygenfile` line. When that happens, the generated documentation for the functions does not have links. Removing `doxygenfile` fixes the problem.

# Expected behavior
I expected `doxygenfile` to not generate declarations for the sections that are not specified under `:sections:`.

# Versions
sphinx-3.4.1
breathe-4.25.1
doxygen-1.8.20
python-3.9.0

**Comment by inconstxpr** [comment]
Just added `:no-link:`, still having the same problems.

**Comment by vermeeren** [comment]
@inconstxpr It still shows the main function/thing name prior to the `briefdescription` and `detaileddescription` if I remember correctly, this means the function itself is declared anyway.

I don't know a quick fix for this except declaring it only once or enforcing the order of declarations via RST, which is I admit rather ugly. Perhaps some other issues have some other useful information, duplicate IDs problem exists for a long time now. https://github.com/michaeljones/breathe/issues?q=is%3Aissue+duplicate+ID

I think there was some talk about handling this better in Sphinx side and then exposing something additional to Breathe, but I cannot remember this well.

**Comment by inconstxpr** [comment]
@vermeeren You mean `doxygenfile` shows the functions or file name? It doesn't for me, this:
```
Dynamic Array
=============

.. doxygenfile:: DynamicArray.h
        :sections: briefdescription detaileddescription

**Header:** `DynamicArray.h`
```
Turns into the following:
![image](https://user-images.githubusercontent.com/55115875/104019935-29eb4f80-519b-11eb-8a9e-c54bca4aee8f.png)
Which is why I found it quite strange that generates declarations for everything.

Changing the order does "fix" the issue, since then the `doxigenfunction` directives become the first declaration, and the duplication happens on `doxygenfile` where it no longer matters since I'm not using it to document the functions. However, having the file description at the end of the doc page doesn't make a lot of sense to me.

Maybe there's an alternate way to render the file description?

**Comment by vermeeren** [comment]
@inconstxpr If that's the only output from the directive then I remembered wrong, in that case it doesn't make sense for Breathe to declare it. I guess the section logic runs rather late during rendering, at which point things are already declared. I lack the time to dive into this myself currently, you may want to check #501 to investigate.

**Comment by inconstxpr** [comment]
@vermeeren I'm afraid I can't find anything on #501 that looks like a possible cause. I hope someone runs into this and manages to catch something I'm missing.

**Comment by inconstxpr** [comment]
@vermeeren I want to try and take a look, but I would appreciate some pointers if possible.

I'm trying to generate an executable so I can test my local changes on my Doxygen XML, so I'm doing:
```
python .\setup.py build
python .\setup.py install --home <some directory>
<some directory>\bin\sphinx-build.exe -b html -Dbreathe_projects.<project>=<doxygen XML folder> <conf.py location> <output>
```

But in doing so, my local changes don't seem to matter. Am I missing something?

**Comment by vermeeren** [comment]
@inconstxpr build and install should not be needed. What I usually run during development is, from the root of the breathe project :

```sh
export PYTHONPATH="$PWD:$PYTHONPATH"
```

That way it uses breathe from the git repo directly. I am not sure how to set environment variables of Windows, but doing that should do the trick too.

**Comment by vermeeren** [comment]
Just released v4.26.1 containing the fix.
